django/db/views.py

```diff
@@ -2111,6 +2123,16 @@ def small_molecules_for_dataset(self,dataset_id):
 #        # view of the small molecule table
 #        logger.info(str(('small molecules for dataset',dataset_id,len(queryset))))
 #        return queryset
+
+    # Need a second distinction, because we'll only show the sm column 
+    # if there are datarecord entries with cells
+    def has_small_molecules_for_datarecords(self, dataset_id):
+        sql = ( 'SELECT count (smallmolecule_id) FROM db_datarecord dr '
```

Comment:
Thanks, I actually have realized this about count(*) vs. exists.
Since this format begs for an explanation from me, it is:
- incremental code feature being added to an existing set of similar features
- following the pattern from before
- presumably, I was unaware of this or overlooked it during the original implementation
- the optimization is minimal (however, this is a potential bug that will manifest in the future as db size grows).
Based on your input and these points, I think it's worth going back over all instances and fixing.  Will update and fix.